src/transfers-history/client.ts

```diff
@@ -0,0 +1,132 @@
+import EventEmitter from "events";
+import { clientConfig } from "./config";
+import { SpokePoolEventsQueryService } from "./services/SpokePoolEventsQueryService";
+import { Logger, LogLevel } from "./adapters/logger";
+import { providers } from "ethers";
+import { SpokePool, SpokePool__factory } from "@across-protocol/contracts-v2";
+import { ChainId } from "./adapters/web3/model";
+import { SpokePoolEventsQuerier } from "./adapters/web3";
+import { TransfersRepository } from "./adapters/db/transfers-repository";
+
+export enum TransfersHistoryEvent {
+  TransfersUpdated = "TransfersUpdated",
+}
+
+export type TransfersUpdatedEventListenerParams = {
+  depositorAddr: string;
+  filledTransfersCount: number;
+  pendingTransfersCount: number;
+};
+
+export type TransfersUpdatedEventListener = (params: TransfersUpdatedEventListenerParams) => void;
+export type TransfersHistoryClientEventListener = TransfersUpdatedEventListener;
+export type TransfersHistoryClientParams = {
+  chains: {
+    chainId: ChainId;
+    providerUrl: string;
+  }[];
+  pollingIntervalSeconds?: number;
+};
+
+export class TransfersHistoryClient {
+  private eventEmitter = new EventEmitter();
+  private web3Providers: Record<ChainId, providers.Provider> = {};
+  private spokePoolInstances: Record<ChainId, SpokePool> = {};
+  private eventsQueriers: Record<ChainId, SpokePoolEventsQuerier> = {};
+  private eventsServices: Record<string, Record<ChainId, SpokePoolEventsQueryService>> = {};
+  private pollingIntervalSeconds: number = 15;
+  private pollingTimers: Record<string, NodeJS.Timer[]> = {};
+
+  constructor(
+    config: TransfersHistoryClientParams,
+    private logger = new Logger(),
+    private transfersRepository = new TransfersRepository()
+  ) {
+    if (config.pollingIntervalSeconds) {
+      this.pollingIntervalSeconds = config.pollingIntervalSeconds;
+    }
+
+    for (const chain of config.chains) {
+      clientConfig.web3ProvidersUrls[chain.chainId] = chain.providerUrl;
+      this.web3Providers[chain.chainId] = new providers.JsonRpcProvider(chain.providerUrl);
+      this.spokePoolInstances[chain.chainId] = SpokePool__factory.connect(
+        clientConfig.spokePools[chain.chainId].addr,
+        this.web3Providers[chain.chainId]
+      );
+      this.eventsQueriers[chain.chainId] = new SpokePoolEventsQuerier(
+        this.spokePoolInstances[chain.chainId],
+        undefined,
+        this.logger
+      );
+    }
+  }
+
+  public setLogLevel(level: LogLevel) {
+    this.logger.setLevel(level);
+  }
+
+  public async startFetchingTransfers(depositorAddr: string) {
+    const timers = this.pollingTimers[depositorAddr];
+    if (timers) throw new Error(`Address ${depositorAddr} is already monitored`)
+
+    this.initSpokePoolEventsQueryServices(depositorAddr);
+    this.getEventsForDepositor(depositorAddr);
+    const timer = setInterval(() => {
+      this.getEventsForDepositor(depositorAddr);
+    }, this.pollingIntervalSeconds * 1000);
```

Comment:
Why does this client include polling?

If this is meant for the frontend, doesn't this go against the philosophy of allowing the frontend to manage the frequency and orchestration of updates in hooks? Otherwise, don't you end up with problems where you can update midway through render cycles?

Reply:
We can disable the polling part (or add a config parameter for enabling it), but my vision for the FE page is to make it more interactive by monitoring and updating the deposits table in real time, not showing just a static table. 
I see this client as a "backend" which hides the implementation details and expose the simplest public interface to the FE. 

I don't think we can end up with these kind of problems because the FE updates its state by calling the getTransfers() functions, like an API endpoint for example. The events emitted are just to inform the FE and the user that new data is available